decorator_allowed_groups/models/decorator.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (C) 2021-Today: GRAP (<http://www.grap.coop/>)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api
+from odoo.exceptions import AccessError
+
+
+def allowed_groups(*group_xml_ids):
+    """ Return a decorator that specifies group(s)
+        to which the user must belong in order to perform
+        this function.
+        - if the user does not belong to any group,
+        an AccessError will be raised if the function is called.
+        - Also in the generated views, the according button
+        will be hidden if the user doesn't belong to the group(s).
+
+            @api.allowed_groups(
```

Comment:
@api.groups should be enough and is aligned with the current declaration in the xml

Reply:
Matter of taste, maybe. I followed the @lmignon's recommandation.
https://github.com/OCA/server-tools/issues/944#issuecomment-336069286

I have no opinion on the subject.

Reply:
In my opinion allowed_groups is much clearer and makes the code self documenting.